src/components/Trade/TradeInput.tsx

```diff
@@ -413,36 +415,47 @@ export const TradeInput = () => {
     [sellAmountCryptoPrecision, buyAmountCryptoPrecision, isTradeQuotePending],
   )
 
-  const hasSufficientProtocolFeeBalances = useMemo(() => {
-    if (protocolFees === undefined || tradeQuoteArgs === undefined || !buyAssetAccountId)
-      return false
+  const insufficientBalanceProtocolFeeMeta = useMemo(() => {
+    if (protocolFees === undefined || tradeQuoteArgs === undefined || !buyAssetAccountId) return
 
     const isBuyingOsmoWithOmosisSwapper =
       activeSwapper?.name === SwapperName.Osmosis && buyAsset.assetId === osmosisAssetId
 
     // This is an oversimplification where protocol fees are assumed to be only deducted from
     // account IDs corresponding to the sell asset account number and protocol fee asset chain ID.
     // Later we'll need to handle protocol fees payable from the buy side.
-    return Object.entries(protocolFees)
-      .filter(([_assetId, protocolFee]) => protocolFee.requiresBalance)
-      .every(([assetId, protocolFee]) => {
+    const insufficientBalanceProtocolFees = Object.entries(protocolFees).filter(
+      ([assetId, protocolFee]: [AssetId, ProtocolFee]) => {
+        if (!protocolFee.requiresBalance) return false
+
         // TEMP: handle osmosis protocol fee payable on buy side for specific case until we implement general solution
         const accountId = isBuyingOsmoWithOmosisSwapper
           ? buyAssetAccountId
           : portfolioAccountIdByNumberByChainId[tradeQuoteArgs.accountNumber][
               protocolFee.asset.chainId
             ]
         const balanceCryptoBaseUnit = portfolioAccountBalancesBaseUnit[accountId][assetId]
-        return bnOrZero(balanceCryptoBaseUnit).gte(protocolFee.amountCryptoBaseUnit)
-      })
+        return bnOrZero(balanceCryptoBaseUnit).lt(protocolFee.amountCryptoBaseUnit)
+      },
+    )
+
+    if (!insufficientBalanceProtocolFees[0]) return
+
+    const protocolFee = insufficientBalanceProtocolFees[0][1]
```

Comment:
I'd love to see a comment here explaining why we are using only the first element, as it's not immediately obvious (that the UI can currently only show one error message at a time, so we show the first).

Reply:
done and done